src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx

```diff
@@ -0,0 +1,147 @@
+        direction="row"
+        gap="xsmall"
+        pad={{ vertical: 'xsmall', horizontal: 'small' }}
+        margin="xsmall"
+        background="brand"
+        round="xsmall"
+        width="small"
+        height="calc(100% - 10px)"
+      >
+        <Text size="small">
+          {group.name} ({group.members?.length})
+        </Text>
+        <Box round="full" margin={{ left: 'xsmall' }}>
+          <FormClose size="small" style={{ width: '12px', height: '12px' }} />
+        </Box>
+      </Box>
+    </Button>
+  );
+};
+
+export interface AddGroupToRoomDialogProps {
+  open: boolean;
+  onClose: () => void;
+  roomSlug: string;
+}
+
+const AddGroupToRoomDialog = ({ open, onClose, roomSlug }: AddGroupToRoomDialogProps) => {
+  const intl = useIntl();
+  const controller = useController();
+  const { data: groups, isLoading: groupsLoading } = useQuery('groups', controller.getGroups);
+  const { mutate, isLoading } = useMutation(controller.addGroupsToRoom, {
+    onSuccess: () => onClose(),
+  });
+  const [selected, setSelected] = useState<number[]>([]);
+
+  const handleUnselect = (id: string) => {
+    setSelected(selected.filter((groupIndex) => groups?.[groupIndex]?.id !== id));
+  };
+
+  if (!open) return null;
+  return (
+    <Layer
+      id="confirmDelete"
+      position="center"
+      onClickOutside={onClose}
+      onEsc={onClose}
+      role="dialog"
+    >
+      <Box pad="medium" width="large">
+        <Heading level={3} size="small" color="brand">
+          {intl.formatMessage(messages.AddGroupToRoomDialogTitle)}
+        </Heading>
+        {groupsLoading ? (
+          <WaitingRow />
+        ) : (
+          <Select
+            closeOnChange={false}
+            multiple
+            disabled={isLoading}
+            value={
+              <Box wrap direction="row">
+                {selected && selected.length > 0 ? (
+                  selected.map((groupIndex) => renderGroup(groups?.[groupIndex], handleUnselect))
+                ) : (
+                  <Box pad="small">{intl.formatMessage(messages.selectGroupsToAdd)}</Box>
+                )}
+              </Box>
+            }
+            options={(groups || []).map((g) => g.name)}
+            selected={selected}
+            onChange={({ selected: newSelected }) => {
+              setSelected(newSelected);
+            }}
+          />
+        )}
+        <Box direction="row" justify="end" margin={{ top: 'medium' }}>
+          <Button
+            label={intl.formatMessage(messages.cancelLabel)}
+            onClick={onClose}
+            onFocus={(event: React.FocusEvent) => event.stopPropagation()}
+            margin={{ right: 'small' }}
+          />
+          <LoadingButton
+            label={intl.formatMessage(messages.addGroupsLabel, { numberSelected: selected.length })}
+            primary
+            onClick={() =>
+              mutate({ roomSlug, groupIds: selected.map((index) => groups?.[index]?.id as string) })
+            }
```

Comment:
I think you should put this in a function.

Reply:
Indeed.